library/Icingadb/Widget/Detail/RedundancyGroupDetail.php

```diff
@@ -0,0 +1,122 @@
+use Icinga\Module\Icingadb\Common\Database;
+use Icinga\Module\Icingadb\Common\Links;
+use Icinga\Module\Icingadb\Model\DependencyNode;
+use Icinga\Module\Icingadb\Model\UnreachableParent;
+use Icinga\Module\Icingadb\Redis\VolatileStateResults;
+use Icinga\Module\Icingadb\Widget\ItemList\DependencyNodeList;
+use Icinga\Module\Icingadb\Hook\ExtensionHook\ObjectDetailExtensionHook;
+use Icinga\Module\Icingadb\Widget\ShowMore;
+use ipl\Html\BaseHtmlElement;
+use ipl\Html\HtmlElement;
+use ipl\Html\Text;
+use ipl\Stdlib\Filter;
+
+class RedundancyGroupDetail extends BaseHtmlElement
+{
+    use Auth;
+    use Database;
+
+    protected $group;
+
+    protected $defaultAttributes = [
+        'class' => ['redundancygroup-detail'],
+        'data-pdfexport-page-breaks-at' => 'h2'
+    ];
+
+    protected $tag = 'div';
+
+    public function __construct($group)
+    {
+        $this->group = $group;
+    }
+
+    protected function createExtensions(): array
+    {
+        return ObjectDetailExtensionHook::loadExtensions($this->group);
+    }
+
+    /**
+     * Create a list of root problems if group fails
+     *
+     * @return ?BaseHtmlElement[]
+     */
+    protected function createRootProblems(): ?array
+    {
+        if (! $this->group->state->failed) {
+            return null;
+        }
+
+        $rootProblems = UnreachableParent::on($this->getDb(), $this->group)
+            ->with([
+                'host',
+                'host.state',
+                'host.icon_image',
+                'host.state.last_comment',
+                'service',
+                'service.state',
+                'service.icon_image',
+                'service.state.last_comment',
+                'service.host',
+                'service.host.state'
+            ])
+            ->setResultSetClass(VolatileStateResults::class)
+            ->orderBy([
+                'host.state.severity',
+                'host.state.last_state_change',
+                'service.state.severity',
+                'service.state.last_state_change'
+            ], SORT_DESC);
+
+        $this->applyRestrictions($rootProblems);
+
+        return [
+            HtmlElement::create('h2', null, Text::create(t('Root Problems'))),
+            (new DependencyNodeList($rootProblems))->setEmptyStateMessage(
+                t('You are not authorized to view these objects.')
+            )
+        ];
+    }
+
+    protected function createGroupMembers()
+    {
+        $membersQuery = DependencyNode::on($this->getDb())
+            ->with([
+                'host',
+                'host.state',
+                'service',
+                'service.state',
+                'service.host',
+                'service.host.state'
+            ])
+            ->filter(Filter::equal('child.redundancy_group.id', $this->group->id))
+            ->limit(3)
+            ->peekAhead();
+
+        $this->applyRestrictions($membersQuery);
+
+        $members = $membersQuery->execute();
```

Comment:
don't execute unless you have to

Reply:
The `showMore` class requires `ResultSet` instance as parameter.

Reply:
Ah, you're right :)